File: ViewBase/Source/Converters/DoubleStringConverter.vb
Imports System.Globalization
Imports System.Windows.Data

Public Class DoubleStringConverter
    Implements IValueConverter

    Public Function Convert(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.Convert
        If IsNumeric(value) Then
            Dim dbl = CType(value, Double)
            If parameter IsNot Nothing Then
                Return dbl.ToString(parameter.ToString)
            End If
            Return dbl.ToString()
        End If
        Return "0"
    End Function

    Public Function ConvertBack(value As Object, targetType As Type, parameter As Object, culture As CultureInfo) As Object Implements IValueConverter.ConvertBack
        If IsNumeric(value) Then
            Dim str = CType(value, String)
            Dim doubleValue As Double
            Dim ok = Double.TryParse(str, NumberStyles.Any, CultureInfo.CurrentCulture, doubleValue)
            If ok Then
                Return doubleValue
            End If
        End If
        Return 0
    End Function
End Class

File: CoreView/src/Converters/DoubleStringConverter.cs
using Microsoft.VisualBasic;
using System;
using System.Globalization;
using System.Windows.Data;

namespace Wells.CoreView.Converters
{
    public class DoubleStringConverter : IValueConverter
    {
        object IValueConverter.Convert(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (Information.IsNumeric(value))
            {
                var dbl = Convert.ToDouble(value);
                if (parameter != null)
                    return dbl.ToString(parameter.ToString());
                return dbl.ToString();
            }
            return "0";
        }

        object IValueConverter.ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
        {
            if (Information.IsNumeric(value))
            {
                var str = (string)value;
                double doubleValue;
                bool ok = double.TryParse(str, NumberStyles.Any, CultureInfo.CurrentCulture, out doubleValue);
                if (ok)
                    return doubleValue;
            }
            return 0.0;
        }
    }
}
